### PR TITLE
Update Changelog for 2.4

### PR DESCRIPTION
**Pull request by daschuer**
This adds another 304 lines to the changelog. 
Thank you for all the work and love that is hidden behind each line! 

This is the raw version of the Changelog with all PRs and almost all Issues referenced. 
I like to merge it to make the next meld down step to a hand-able size visible in a diff. 

Please check also the rendered md version.

### CODE REVIEW

**Comment by mxmilkiib** [comment]
Is missing #12376

**Comment by ronso0** [comment]
> Is missing #12376

No, that was for `main`, not `2.4`

**Comment by mxmilkiib** [comment]
Aaah, so all new features go to main, so it's just in 2.5-alpha onward? I wonder how much there is to go wrong with it, but cool, fair enough if there is a process to be stuck to.

**Comment by daschuer** [comment]
At least the raw version I have started with was auto-generated by the attached script. 
All the sorting and matching with issue numbers is a manual process, which can go wrong. 

In the next step after merge we will remove entries not important to users. That can also go wrong. 
Help is welcome.

**Review by ywwg — APPROVED** [review]
lgtm -- we can create a reduced list from this for the blog post and do language correction and refinement there.

File: CHANGELOG.md [hunk 1]
```diff
@@ -360,26 +438,51 @@
 
 ### Controller Mappings
 
-* Novation Launchpad: Update controller scripts [#2600](https://github.com/mixxxdj/mixxx/pull/2600)
-* Roland DJ-505: Make blinking lights blink in sync and other improvements [#4159](https://github.com/mixxxdj/mixxx/pull/4159) [#4517](https://github.com/mixxxdj/mixxx/pull/4517)
 * Behringer DDM4000 mixer: Update controller mapping [#4262](https://github.com/mixxxdj/mixxx/pull/4262) [#4799](https://github.com/mixxxdj/mixxx/pull/4799)
-* Numark DJ2GO2 Touch: Fix sampler, hotcue, beatloop buttons [#4287](https://github.com/mixxxdj/mixxx/pull/4287)
-* Denon MC6000MK2: Improve mapping code [#4385](https://github.com/mixxxdj/mixxx/pull/4385)
+* Denon MC6000MK2: Improve mapping code [#4385](https://github.com/mixxxdj/mixxx/pull/4385) [#11792](https://github.com/mixxxdj/mixxx/pull/11792)
```
Comment (ronso0) [inline CHANGELOG.md:443]: both PRs went into 2.3
```suggestion
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -360,26 +438,51 @@
...
+* Hercules DJControl MIX: New mapping [#11279](https://github.com/mixxxdj/mixxx/pull/11279)
+* Hercules DJ Console RMX: Replace not defined CO name pitch_reset by pitch_set_default [#12441](https://github.com/mixxxdj/mixxx/pull/12441)
+* Korg nanoKONTROL2: Removed Mixco scripts [#2682](https://github.com/mixxxdj/mixxx/pull/2682)
+* Korg nanoKONTROL2: Don't try to configure more than 4 main decks [#12322](https://github.com/mixxxdj/mixxx/pull/12322) [#12317](https://github.com/mixxxdj/mixxx/issues/12317)
+* MAudio Xponent: Removed along with Mixco scripts [#2682](https://github.com/mixxxdj/mixxx/pull/2682)
+* MIDI4lights: Give beginTimer callbacks the anonymous function expression form [#12048](https://github.com/mixxxdj/mixxx/pull/12048) [#12042](https://github.com/mixxxdj/mixxx/issues/12042)
```
Comment (ronso0) [inline CHANGELOG.md:449]: The issues is still open, the PR did likely not fix it (relying on @Swiftb0y's [comment](https://github.com/mixxxdj/mixxx/pull/12048#pullrequestreview-1681532979) here)
```suggestion
* MIDI4lights: Give beginTimer callbacks the anonymous function expression form [#12048](https://github.com/mixxxdj/mixxx/pull/12048)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -360,26 +438,51 @@
...
+* Numark MixTrack Pro 3: Fix beginTimer callback syntax [#12401](https://github.com/mixxxdj/mixxx/pull/12401) [#12369](https://github.com/mixxxdj/mixxx/issues/12369)
+* Pioneer DDJ-FLX4: New mapping base on DDJ-400 [#11245](https://github.com/mixxxdj/mixxx/pull/11245)
+* Roland DJ-505: Make blinking lights blink in sync and other improvements [#4159](https://github.com/mixxxdj/mixxx/pull/4159) [#4517](https://github.com/mixxxdj/mixxx/pull/4517)
+* Traktor Kontrol S2 MK1: Add calibration and refactor [#11237](https://github.com/mixxxdj/mixxx/pull/11237)
+* Traktor Kontrol S2 MK2 fix loaded chain preset CO [#11823](https://github.com/mixxxdj/mixxx/pull/11823) [#10667](https://github.com/mixxxdj/mixxx/issues/10667)
+* Traktor Kontrol S2 MK3: Use FX select buttons to set quick effect presets
+  [#11702](https://github.com/mixxxdj/mixxx/pull/11702)
+* Traktor Kontrol S3: script improvements, vanilla-like FX behavior, control initialization, better scratching, and more
+  [#11199](https://github.com/mixxxdj/mixxx/pull/11199)
+  [#4646](https://github.com/mixxxdj/mixxx/pull/4646)
```
Comment (ronso0) [inline CHANGELOG.md:462]: that one went into 2.3
```suggestion
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -360,26 +439,51 @@
...
-* Korg Nanokontrol 2: Removed along with Mixco Scripts [#2682](https://github.com/mixxxdj/mixxx/pull/2682)
-* MAudio Xponent: Removed along with Mixco Scripts [#2682](https://github.com/mixxxdj/mixxx/pull/2682)
+* Various mappings: Fix `waveform_zoom` ranges [#12393](https://github.com/mixxxdj/mixxx/pull/12393)
 
 ### Controller Backend
 
 * Never raise a fatal error if a controller mapping tries access a non-existent control object [#2947](https://github.com/mixxxdj/mixxx/pull/2947)
-* Add support to access HID FeatureReports [#3051](https://github.com/mixxxdj/mixxx/pull/3051)
+* Add support to access HID FeatureReports
+  [#3051](https://github.com/mixxxdj/mixxx/pull/3051)
```
Comment (ronso0) [inline CHANGELOG.md:475]: was merged to master in 2020
```suggestion
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -394,22 +498,24 @@
 * Fix segfault during Mixxx shutdown due to a stale controller connection [#4476](https://github.com/mixxxdj/mixxx/pull/4476) [#10553](https://github.com/mixxxdj/mixxx/issues/10553)
 * Components JS: Fix syncbutton [#4329](https://github.com/mixxxdj/mixxx/pull/4329)
 * Components JS: Add script.posMod for euclidean modulo [#11415](https://github.com/mixxxdj/mixxx/pull/11415)
+* Components JS: make JogWheelBasic correctly switch which deck it controls [#11913](https://github.com/mixxxdj/mixxx/pull/11913) [#11867](https://github.com/mixxxdj/mixxx/issues/11867)
 * Added Trace for the mapping connections, to allow JS profiling [#4766](https://github.com/mixxxdj/mixxx/pull/4766)
 * Fix crash when using `midi.sendShortMsg` and platform vnc [#4635](https://github.com/mixxxdj/mixxx/pull/4635)
```
Comment (ronso0) [inline CHANGELOG.md:503]: that one went into 2.3 before 2.3.6 was released

File: CHANGELOG.md [hunk 6]
```diff
@@ -488,14 +614,17 @@
...
   [#3624](https://github.com/mixxxdj/mixxx/pull/3624)
 * Make beat indicator control behaviour more natural [#3608](https://github.com/mixxxdj/mixxx/pull/3608)
-* Allow skin scaling from preferences [#3960](https://github.com/mixxxdj/mixxx/pull/3960)
-* Invert scroll wheel waveform zoom direction to mach other applications [#4195](https://github.com/mixxxdj/mixxx/pull/4195)
+* Allow skin scaling from preferences
+  [#3960](https://github.com/mixxxdj/mixxx/pull/3960)
+  [#11588](https://github.com/mixxxdj/mixxx/pull/11588)
+  [#11586](https://github.com/mixxxdj/mixxx/issues/11586)
+* Increased pixmapCache size limit and made it dependent on devicePixelRatio (for HiDPI/Retina displays) [#12416](https://github.com/mixxxdj/mixxx/pull/12416)
+* Fix Icon rendering on HiRDPI/retina screens [#12407](https://github.com/mixxxdj/mixxx/pull/12407) [#12361](https://github.com/mixxxdj/mixxx/issues/12361)
```
Comment (ronso0) [inline CHANGELOG.md:627]:
```suggestion
* Fix Icon rendering on HiDPI/retina screens [#12407](https://github.com/mixxxdj/mixxx/pull/12407) [#12361](https://github.com/mixxxdj/mixxx/issues/12361)
```

**Review by ronso0 — CHANGES_REQUESTED** [review]
this is my first run, stopped above the Waveforms section

**Comment by ronso0** [comment]
Found only small issues.
Please remove
`[#4255](https://github.com/mixxxdj/mixxx/pull/4455)`

**Comment by daschuer** [comment]
Thank you for the detailed review!

**Comment by daschuer** [comment]
We had a brief discussion how to proceed with this bulky long changelog. A consensus was to not put much extra hours to it for polishing, because no user will read it line by line. In earlier entries we have removed developer topics. I think I will at least do this to match the style of previous entries after merge. Ideas?

**Comment by ronso0** [comment]
Sute, except mapping dev topics I assume.

File: tools/createChangelog.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import git
```
Comment (ywwg) [inline tools/createChangelog.py:1]: can you add a short docstyle comment to describe what this script does?  (does it update the changelog or create it from scratch? when should it be used? etc)

**Review by ywwg — APPROVED** [review]

File: tools/createChangelog.py (hunk 7 above)
Comment (daschuer) [reply tools/createChangelog.py]: Done.

**Comment by daschuer** [comment]
Ready.

**Review by Swiftb0y — APPROVED** [review]
LGTM. waiting for @ronso0 to approve or resolve.

File: CHANGELOG.md [hunk 8]
```diff
@@ -548,7 +735,16 @@
...
+* Effects: Allow clearing chains with empty '---' preset [#10859](https://github.com/mixxxdj/mixxx/pull/10859) [#10777](https://github.com/mixxxdj/mixxx/issues/10777)
+* Effectchain tooltip [#11902](https://github.com/mixxxdj/mixxx/pull/11902) [#10605](https://github.com/mixxxdj/mixxx/issues/10605)
+* Effects: Center Super knob when loading empty (QuickEffect) chain preset [#12320](https://github.com/mixxxdj/mixxx/pull/12320)
+* Effects: Read effect data for decks added after initial setup [#12282](https://github.com/mixxxdj/mixxx/pull/12282) [#12277](https://github.com/mixxxdj/mixxx/issues/12277)
+* Effects: Fix deletion of effect chain presets created in current session [#11886](https://github.com/mixxxdj/mixxx/pull/11886)
+* Effects: Don't reset "super" and "mix" knob on startup [#11781](https://github.com/mixxxdj/mixxx/pull/11781) [#11773](https://github.com/mixxxdj/mixxx/issues/11773)
+* Effects: Re-add empty chain preset when resetting to defaults [#11705](https://github.com/mixxxdj/mixxx/pull/11705)
+* Effects: Add a missing early return [#11809](https://github.com/mixxxdj/mixxx/pull/11809) [#111808](https://github.com/mixxxdj/mixxx/issues/11808)
```
Comment (ronso0) [inline CHANGELOG.md:745]:
```suggestion
* Allow clearing chains with empty '---' preset [#10859](https://github.com/mixxxdj/mixxx/pull/10859) [#10777](https://github.com/mixxxdj/mixxx/issues/10777)
* Show Effectchain tooltip [#11902](https://github.com/mixxxdj/mixxx/pull/11902) [#10605](https://github.com/mixxxdj/mixxx/issues/10605)
* Center Super knob when loading empty (QuickEffect) chain preset [#12320](https://github.com/mixxxdj/mixxx/pull/12320)
* Read effect data for decks added after initial setup [#12282](https://github.com/mixxxdj/mixxx/pull/12282) [#12277](https://github.com/mixxxdj/mixxx/issues/12277)
* Fix deletion of effect chain presets created in current session [#11886](https://github.com/mixxxdj/mixxx/pull/11886)
* Don't reset "super" and "mix" knob on startup [#11781](https://github.com/mixxxdj/mixxx/pull/11781) [#11773](https://github.com/mixxxdj/mixxx/issues/11773)
* Re-add empty chain preset when resetting to defaults [#11705](https://github.com/mixxxdj/mixxx/pull/11705)
* Add a missing early return [#11809](https://github.com/mixxxdj/mixxx/pull/11809) [#111808](https://github.com/mixxxdj/mixxx/issues/11808)
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -620,6 +821,22 @@
   [#3113](https://github.com/mixxxdj/mixxx/pull/3113)
   [#3089](https://github.com/mixxxdj/mixxx/pull/3089)
   [#3545](https://github.com/mixxxdj/mixxx/pull/3545)
+* Windows packaging: Use Azure for signing for exe, msi and all dlls with timestamp and sha256
```
Comment (ronso0) [inline CHANGELOG.md:824]:
```suggestion
* Windows packaging: Use Azure for signing exe, msi and all dlls with timestamp and sha256
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -1025,6 +1319,38 @@
...
+### Known issues
+
+* Volume / Loudness spikes on Windows with M4A/AAC files.
+  Last known working version is Windows 10 build 17763.
+  Affected versions are Windows 10 build 19041 and Windows 11 build 22000.
+  A fix is promised in 2024-02 with Windows 11 build 26016.
+  [#12289](https://github.com/mixxxdj/mixxx/issues/12289)
+  [#11094](https://github.com/mixxxdj/mixxx/issues/11094)
+* Different, language depending library sorting on macOS
+  [#12517](https://github.com/mixxxdj/mixxx/issues/12517)
```
Comment (ronso0) [inline CHANGELOG.md:1353]: I remember we came across an issue recently that we wanted to add to 'known issues' (but we didn't think of this place). I added the Changelog label and added it to the 2.4 milestone, but I can't find it anymore...

Maybe someone else remembers what that was.
I think @fwcd and/or @m0dB were involved.

**Review by ronso0 — COMMENTED** [review]
LGTM
Let's remove the redundant 'Effects' prefix.

**Comment by ronso0** [comment]
Thank you, looks good!

**Pull request merged**